pkg/services/sqlstore/session/session.go

```diff
@@ -0,0 +1,104 @@
+package session
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type Session interface {
+	Get(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+	Exec(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	DriverName() string
+}
+
+type SessionDB struct {
+	sqlxdb *sqlx.DB
+}
+
+func GetSession(sqlxdb *sqlx.DB) *SessionDB {
+	return &SessionDB{sqlxdb: sqlxdb}
+}
+
+func (gs *SessionDB) Get(ctx context.Context, dest interface{}, query string, args ...interface{}) error {
+	return gs.sqlxdb.GetContext(ctx, dest, gs.sqlxdb.Rebind(query), args...)
+}
+
+func (gs *SessionDB) Select(ctx context.Context, dest interface{}, query string, args ...interface{}) error {
+	return gs.sqlxdb.SelectContext(ctx, dest, gs.sqlxdb.Rebind(query), args...)
+}
+
+func (gs *SessionDB) Exec(ctx context.Context, query string, args ...interface{}) (sql.Result, error) {
+	return gs.sqlxdb.ExecContext(ctx, gs.sqlxdb.Rebind(query), args...)
+}
+
+func (gs *SessionDB) DriverName() string {
+	return gs.sqlxdb.DriverName()
+}
+
+func (gs *SessionDB) Beginx() (*SessionTx, error) {
+	tx, err := gs.sqlxdb.Beginx()
+	return &SessionTx{sqlxtx: tx}, err
+}
+
+func (gs *SessionDB) WithTransaction(ctx context.Context, callback func(*SessionTx) error) error {
```

Comment:
we might need a shared transaction in context later on, not implemented it yet in this pr